pkg/discovery/dtofactory/container_dto_builder.go

```diff
@@ -0,0 +1,162 @@
+			if err != nil {
+				glog.Errorf("failed to create commoditiesSold for container[%s]: %v", name, err)
+				continue
+			}
+			ebuilder.SellsCommodities(commoditiesSold)
+
+			//2. commodities bought
+			commoditiesBought, err := builder.getCommoditiesBought(podId, containerId, nodeCPUFrequency)
+			if err != nil {
+				glog.Errorf("failed to create commoditiesBought for container[%s]: %v", name, err)
+				continue
+			}
+			provider := sdkbuilder.CreateProvider(proto.EntityDTO_CONTAINER_POD, podId)
+			ebuilder.Provider(provider).BuysCommodities(commoditiesBought)
+
+			//3. set properties
+			if !util.Monitored(pod) {
+				ebuilder.Monitored(false)
+			}
+
+			//4. build entityDTO
+			dto, err := ebuilder.Create()
+			if err != nil {
+				glog.Errorf("faild to builld Container[%s] entityDTO: %v", name, err)
+			}
+
+			result = append(result, dto)
+		}
+	}
+
+	return result, nil
+}
+
+//vCPU, vMem, Application are sold by Container to Application
+func (builder *containerDTOBuilder) getCommoditiesSold(containerKey string, cpuFrequency float64) ([]*proto.CommodityDTO, error) {
+	var result []*proto.CommodityDTO
+
+	//1. vCPU & vMem
+	converter := NewConverter().Set(func(input float64) float64 { return input * cpuFrequency }, metrics.CPU, metrics.CPUProvisioned)
+
+	attributeSetter := NewCommodityAttrSetter()
+	attributeSetter.Add(func(commBuilder *sdkbuilder.CommodityDTOBuilder) { commBuilder.Resizable(true) }, metrics.CPU, metrics.Memory)
+
+	commodities, err := builder.getResourceCommoditiesSold(task.ContainerType, containerKey, commoditySold, converter, attributeSetter)
+	if err != nil {
+		return nil, err
+	}
+	result = append(result, commodities...)
+
+	//2. Application
+	appCommodity, err := sdkbuilder.NewCommodityDTOBuilder(proto.CommodityDTO_APPLICATION).
+		Key(containerKey).
+		Capacity(applicationCommodityDefaultCapacity).
+		Create()
+	if err != nil {
+		return nil, err
+	}
+	result = append(result, appCommodity)
+
+	return result, nil
+}
+
+// vCPU, vMem and VMPMAccess are bought by Container from Pod;
+// the VMPMAccess is to bind the container to the hosting pod.
+func (builder *containerDTOBuilder) getCommoditiesBought(podId, containerId string, cpuFrequency float64) ([]*proto.CommodityDTO, error) {
+	var result []*proto.CommodityDTO
+
+	//1. vCPU & vMem
+	converter := NewConverter().Set(func(input float64) float64 { return input * cpuFrequency }, metrics.CPU, metrics.CPUProvisioned)
+
+	attributeSetter := NewCommodityAttrSetter()
+	attributeSetter.Add(func(commBuilder *sdkbuilder.CommodityDTOBuilder) { commBuilder.Resizable(true) }, metrics.CPU, metrics.Memory)
+
+	commodities, err := builder.getResourceCommoditiesBought(task.ContainerType, containerId, commodityBought, converter, attributeSetter)
+	if err != nil {
+		return nil, err
+	}
+	result = append(result, commodities...)
+
+	//2. VMPMAccess
+	podAccessComm, err := sdkbuilder.NewCommodityDTOBuilder(proto.CommodityDTO_VMPM_ACCESS).
```

Comment:
We might be able to use Moveable flag to control this. Again, let's talk to Wei.